petab/visualize/plotting.py

```diff
@@ -0,0 +1,898 @@
+                                             Optional[Real]]] = None,
+                        ylim: Optional[Tuple[Optional[Real],
+                                             Optional[Real]]] = None):
+        """
+        Set axes limits for all subplots. If xlim or ylim or any of the tuple
+        items is None, corresponding limit is left unchanged.
+
+        Parameters
+        ----------
+        xlim: x axis limits
+        ylim: y axis limits
+        """
+        self.xlim = xlim
+        self.ylim = ylim
+
+
+class Figure:
+    def __init__(self, subplots: Optional[List[Subplot]] = None,
+                 size: Tuple = (20, 15),
+                 title: Optional[Tuple] = None):
+        """
+        Visualization specification of a figure. Contains information
+        regarding how data should be visualized.
+
+        Parameters
+        ----------
+        subplots
+        size
+        title
+        """
+
+        # TODO: Isensee measurements table in doc/examples doesn't correspond
+        #       to documentation: observableTransformation and
+        #       noiseDistribution columns replicateId problem
+        # TODO: Should we put in the documentation which combination of fields
+        #  must be unique in the measurement table and add such check?
+        #  obs_id + sim_cond_id + preeq_cod_id (if exists) + time +
+        #  replicate_id (if exists)?
+        self.size = size
+        self.title = title
+        self.subplots = subplots if subplots is not None else []
+
+    @property
+    def num_subplots(self) -> int:
+        return len(self.subplots)
+
+    def add_subplot(self, subplot: Subplot):
+        self.subplots.append(subplot)
+
+    def set_axes_limits(self,
+                        xlim: Optional[Tuple[Optional[Real],
+                                             Optional[Real]]] = None,
+                        ylim: Optional[Tuple[Optional[Real],
+                                             Optional[Real]]] = None):
+        """
+        Set axes limits for all subplots. If xlim or ylim or any of the tuple
+        items is None, corresponding limit is left unchanged.
+
+        Parameters
+        ----------
+        xlim: x axis limits
+        ylim: y axis limits
+        """
+
+        for subplot in self.subplots:
+            subplot.set_axes_limits(xlim, ylim)
+
+    def save_to_tsv(self, output_file_path: str = 'visuSpec.tsv') -> None:
+        """
+        save full Visualization specification table
+
+        Note that datasetId column in the resulting table might have been
+        generated even though datasetId column in Measurement table is missing
+        or is different. Please, correct it manually.
+
+        Parameters
+        ----------
+        output_file_path: File path to which the generated visualization
+                          specification is saved.
+        """
+        # what if datasetIds were generated?
```

Comment:
to be discussed... maybe then also saving an updated version of measurement table, but without replacing the original one?